api/src/controllers/AuthenticationController.ts

```diff
@@ -1,56 +1,58 @@
 import { Request, Response } from "express";
-import AuthService from "../services/AuthService";
-import ErrorController from "./ErrorController";
 import { UserAttributes } from "../utils/types/attributeTypes";
-import {
-  generateSignedJwt,
-  hashPassword,
-  verifyPassword,
-} from "../utils/tokenService";
+import AuthService from "../services/AuthenticationService";
 
-export default class AuthController extends ErrorController {
+type GenericClaimStructure = {
+  iss: string;
+  sub: string;
+  aud: string;
+  exp: number;
+  iat: number;
+};
+
+interface UserLogin {
+  username: string;
+  password: string;
+}
+
+export default class AuthenticationContoller {
   private authService = new AuthService();
 
-  public async signin(req: Request, res: Response) {
+  public async signIn(req: Request, res: Response) {
     try {
-      const data: UserAttributes = req.body;
-
+      const data: UserLogin = req.body;
       const userDetails = await this.authService.getUser(data.username);
-
       if (!userDetails) {
         console.log("User not found - Error 400");
         throw new Error("User not found - Error 400");
       }
-      const isValid = await verifyPassword(userDetails.password, data.password);
+
+      const isValid = await this.authService.passwordVerification(
+        userDetails.password,
+        data.password
+      );
 
       if (!isValid) {
         console.log("Not valid password - Error 403");
         throw new Error("Not valid password - Error 403");
       }
-      const token = await generateSignedJwt(
-        userDetails.id as number,
-        userDetails.scopes
-      );
+
       res.status(200).send({
-        accessToken: token,
-        userId: userDetails.id,
-        email: userDetails.email,
-        firstName: userDetails.firstName,
-        lastName: userDetails.lastName,
+        idToken: this.authService.generateIdToken(userDetails),
+        accessToken: this.authService.generateAccessToken(userDetails.scopes),
       });
     } catch (err) {
       res.send(err);
```

Comment:
C: This is not going to send back correct status for errors, sending `200` each time.

Reply:
Temp work around at the moment but my next task is to do proper error handling for expressjs. Commit: 66504d03b5cd2997feaa51437addf50dc1d6b874